src/Tests/Microsoft.NET.TestFramework/ExponentialRetry.cs

```diff
@@ -34,10 +34,10 @@ public static IEnumerable<TimeSpan> Intervals
             foreach (var t in timer())
```

Comment:
It looks like this will always wait 5 seconds before trying to run a command the first time, which means every single TestCommand will take at least 5 seconds to run.  The first try should happen immediately, and we should only start doing delays if there is a failure.

Reply:
first retry immediately